handler.go

```diff
@@ -194,5 +196,25 @@ func (pset pathConfigSet) find(path string) (pc *pathConfig, subpath string) {
 	if i > 0 && strings.HasPrefix(path, pset[i-1].path+"/") {
 		return &pset[i-1], path[len(pset[i-1].path)+1:]
 	}
-	return nil, ""
+
+	// Slow path, now looking for the longest prefix/shortest subpath i.e.
+	// e.g. given pset ["/", "/abc/", "/abc/def/", "/xyz"/]
+	//  * query "/abc/foo" returns "/abc/" with a subpath of "foo"
+	//  * query "/x" returns "/" with a subpath of "x"
+	lenShortestSubpath := len(path)
+	var bestMatchConfig *pathConfig
+	for i, ps := range pset {
```

Comment:
One optimization I thought of: you don't have to check any configs past `i`, since they will never be a prefix.

Reply:
Nice catch, thank you for this @zombiezen! Cool property of binary search that `mid = (low + high)>>1 ; iff low <= high` so `i` will always be the max of comparator <= query. I'll update that.